src/libstd/io/tempfile.rs

```diff
@@ -78,47 +80,59 @@ pub struct TempDir {
     disarmed: bool
 }
 
+// How many times should we (re)try finding an unused random name? It should be
+// enough that an attacker will run out of luck before we run out of patience.
+const NUM_RETRIES: uint = 1 << 31;
```

Comment:
The OpenBSD source is at http://cvsweb.openbsd.org/cgi-bin/cvsweb/~checkout~/src/lib/libc/stdio/mktemp.c?rev=1.35&content-type=text/plain -- the mktemp_internal function is where it does MAX_INT tries. If it's any consolation, a quick benchmark on my machine suggests it can make and delete about 10K directories per second, so it'll only take 2.5 days to time out ;) I have no firm opinions on what a sensible value is here and if I hadn't read the OpenBSD source I would probably have made it an infinite loop...